### PR TITLE
Basic circulation policies

### PR DESCRIPTION
**Pull request by BadrAly**
NEW: Basic circulation policies
BETTER: Circulation APIs

Co-Authored-by: Aly Badr aly.badr@rero.ch
Co-Authored-by: Gianni Pante gianni.pante@rero.ch

To Test:

get the PR branch
./scripts/bootstrap
./scripts/setup
./run_tests.sh

### CODE REVIEW

File: ui/admin/package.json [hunk 1]
```diff
@@ -33,7 +33,7 @@
     "datatables.net-bs": "^1.10.19",
     "jquery": "^3.3.1",
     "moment": "^2.22.2",
-    "ngx-bootstrap": "3.0.1",
+    "ngx-bootstrap": "^3.1.1",
```
Comment (rerowep) [inline ui/admin/package.json]: Why was this file changed

File: ui/admin/package-lock.json [hunk 2]
```diff
@@ -6739,9 +6739,9 @@
       "dev": true
     },
     "ngx-bootstrap": {
-      "version": "3.0.1",
-      "resolved": "https://registry.npmjs.org/ngx-bootstrap/-/ngx-bootstrap-3.0.1.tgz",
-      "integrity": "sha512-ni91yYtn8ldgf/pxrlwl9lkVcLURGzopSpJnEbbgG1v1EZWTobI8y7J3mx4Kxptkn0EeiQwnLel67G7XJSox4A=="
+      "version": "3.1.2",
```
Comment (rerowep) [inline ui/admin/package-lock.json]: Why was this file changed

File: ui/admin/src/app/users.ts [hunk 3]
```diff
@@ -19,7 +19,6 @@ export class User {
   organisation_pid: string;
   barcode?: string;
   loans?: Loan[];
-
   constructor(obj?: any) {
```
Comment (rerowep) [inline ui/admin/src/app/users.ts]: Why was this file changed

File: rero_ils/modules/loans/utils.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+"""Loans utils."""
+
+
+from datetime import timedelta
+
+from ..items.api import Item
+
+
+def get_default_loan_duration(loan):
```
Comment (jma) [inline rero_ils/modules/loans/utils.py]: is it a duplication of `duration` proroperties of the Item API?

File: rero_ils/modules/patrons/api.py [hunk 5]
```diff
@@ -204,6 +204,11 @@ def roles(self):
             return user.roles
         return []
 
+    @property
+    def barcode(self):
```
Comment (jma) [inline rero_ils/modules/patrons/api.py]: really useful?

File: rero_ils/modules/items/api.py [hunk 6]
```diff
@@ -44,6 +48,59 @@
...
 
+def add_missing_action_params(function):
+    """Add missing action parameters."""
+    @wraps(function)
+    def wrapper(item, *args, **kwargs):
+        kwargs['item_pid'] = item.pid
+        kwargs['document_pid'] = item.dumps().get('document_pid')
+
+        action_required_params = item.action_required_params.copy()
+        from ..patrons.api import current_patron
```
[the remark this answers is not in the document]
Comment (BadrAly) [reply rero_ils/modules/items/api.py]: Circular imports

File: rero_ils/modules/loans/utils.py (hunk 4 above)
Comment (BadrAly) [reply rero_ils/modules/loans/utils.py]: to replace the API. duration is part of the circ policy now

File: rero_ils/modules/patrons/api.py (hunk 5 above)
Comment (BadrAly) [reply rero_ils/modules/patrons/api.py]: nice to have
